src/Generation/generator.py

```diff
@@ -1,62 +1,63 @@
 import json
-import regex
-from llm.connector import connector
+import random
 import importlib
 
-DESCRIPTION_DIRECTORY = "Generation/Descriptions"
-FORM_FILE_NAME = "form.py"
-QUERY_FILE_NAME = "query_text"
-EXAMPLES_FILE_NAME = "examples.json"
-PREDEFINED_FILE_NAME = "predefined.json"
-FORM_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + FORM_FILE_NAME
-QUERY_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + QUERY_FILE_NAME
-EXAMPLES_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + EXAMPLES_FILE_NAME
-PREDEFINED_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + PREDEFINED_FILE_NAME
-FORM_MODUL_NAME = "form"
-FORM_CLASS_NAME = "form"
-EXAMPLES_KEY = "Examples"
-NUMBER_OF_TRIES_TO_GENERATE = 2
+import llm
 
-class Generator:
-    __json_pattern = regex.compile(r'\{(?:[^{}]|(?R))*\}')
+_DESCRIPTION_PATH = 'src/Generation/Descriptions/{type_name}'
+_MODEL_FILE = _DESCRIPTION_PATH + '/model.py'
+_QUERY_FILE = _DESCRIPTION_PATH + '/query_text'
+_EXAMPLES_FILE = _DESCRIPTION_PATH + '/examples.json'
+_PREDEFINED_FILE = _DESCRIPTION_PATH + '/predefined.json'
+_MODEL_MODULE_NAME = "model"
+_MODEL_CLASS_NAME = "Model"
+
+_generator_cache: dict[str, 'Generator'] = {}
+
+def llm_create(type_name: str, count=1, **kwargs) -> list[dict]:
+    if type_name in _generator_cache:
+        generator = _generator_cache[type_name]
+    else:
+        generator = Generator(type_name)
+        _generator_cache[type_name] = generator
+    return generator.generate(count, **kwargs)[1]
```

Comment:
Could we give a warning (logger.warning) if the generation was unsuccessful?

Reply:
Yes, good idea!